src/server.c

```diff
@@ -1094,7 +1123,16 @@ struct nvnc* nvnc_open(const char* address, uint16_t port)
 
 	LIST_INIT(&self->clients);
 
-	self->fd = bind_address(address, port);
+	switch (type) {
+	case ADDRTYPE_TCP:
+		self->fd = bind_address_tcp(address, port);
+		break;
+	case ADDRTYPE_UNIX:
+		self->fd = bind_address_unix(address);
+		break;
+	default:
+		self->fd = -1;
```

Comment:
This situation should never arise if the code is correctly written, so `abort()` will do nicely.